docs/modeling-your-data/modeling-your-data-with-dbt/dbt-advanced-usage/dbt-utils-advanced-operation/index.md

```diff
@@ -7,3 +7,336 @@ sidebar_position: 500
+If there are certain SQL transformations you want to apply to events that are being processed by Snowplow's dbt packages you can leverage the `snowplow__custom_sql` variable to write out custom SQL that will be included in your `base_events_this_run` table, which can then be leveraged for any of your subsequent tables.
+
+The process for this looks slightly different for Redshift & Postgres relative to the other databases. If you use Redshift & Postgres, please skip ahead to the warehouse specific section to understand how that works.
+
+## Utilizing custom contexts or SDEs
+Suppose you have a custom context called `com_mycompany_click_1_0_0` which contains a `click_id` that you want to concat with Snowplow's `domain_sessionid`. You could either do this concatenation when creating your own data models, but if you want to surface this to all of your data models downstream from Snowplow's initial data processing in it's dbt packages, you can add that transformation by adding the following to your `dbt_project.yml`:
+
+```yml
+# dbt_project.yml
+...
+vars:
+    ...
+    snowplow__custom_sql: "CONCAT(com_mycompany_click_1_0_0.click_id, CONCAT('_', domain_sessionid)) as click_session_id"
+    ...
+...
+```
+
+If you'd like to add multiple lines of SQL, you can do that as well by making this string a multi-line string. Any SQL included in the `snowplow__custom_sql` will be found in your `snowplow_base_events_this_run` table, and will be referenced at the end of the select statement, so there's no need to add a trailing comma. Any of the newly created fields can also be passed through to other tables created automatically by Snowplow's dbt packages.
+
+## Utilizing custom contexts or SDEs in Redshift & Postgres
+In Redshift & Postgres, due to the shredded table design (meaning each context is loaded separately into a table), you need to specify which contexts you want to be included in the `snowplow_base_events_this_run` table, which you can do using the `snowplow__entities_or_sdes` variable. The `snowplow__entities_or_sdes` variable expects a list of key:value dictionary (dict) with the following keys:
+
+
+| Key | Description | Example |
+| ----------------------- | ----------------------------- | ----------------------------- |
+| `name` | The name denotes the name of the entity or SDE that you would like to join, which should also be the name of the table that is in your warehouse. | `contexts_com_mycompany_custom_entity_1_0_0` |
+| `prefix` | The prefix that each field in the context will receive. E.g with a prefix of `my_custom` and a field name of `id`, this field will be accessible in the `snowplow_base_events_this_run` table under the `my_custom_id` column. | `my_custom` |
+| `alias` | The alias that is used for the context table join, for reference in your custom SQL queries. | `mce` |
+| `single_entity` | A boolean to say whether this is a single entity or whether there can be multiple for each event. | `true` |
+
```

Comment:
Maybe put a warning here about if single entity is false then you will get duplicate events (by design) in your events this run table, which is unlikely to be what is intended, and it may be easier to join these contexts on in a later model where they are required?